Gifski/util.swift

```diff
@@ -2541,3 +2561,99 @@ extension FileManager {
 		try FileManager.default.copyItem(at: sourceURL, to: destinationURL)
 	}
 }
+
+extension ClosedRange where Bound: AdditiveArithmetic {
+	/// Get the interval between the lower and upper bound.
+	var interval: Bound {
+		return upperBound - lowerBound
+	}
+}
+
+extension ClosedRange {
+	/**
+	Returns true if `self` is a superset of the given range.
+
+	```
+	(1.0...1.5).isSuperset(of: 1.2...1.3)
+	//=> true
+	```
+	*/
+	func isSuperset(of other: ClosedRange<Bound>) -> Bool {
+		return other.isEmpty ||
+			(
+				lowerBound <= other.lowerBound &&
+				other.upperBound <= upperBound
+			)
+	}
+
+	/**
+	Returns true if `self` is a subset of the given range.
+
+	```
+	(1.2...1.3).isSubset(of: 1.0...1.5)
+	//=> true
+	```
+	*/
+	func isSubset(of other: ClosedRange<Bound>) -> Bool {
+		return other.isSuperset(of: self)
+	}
+}
+
+extension ClosedRange where Bound == Double {
+	// TODO: Add support for negative ranges.
+	/**
+	Make a new range where the difference between the lower and upper bound is at least the given amount.
+
+	The use-case for this method is being able to ensure a sub-range inside a range is of a certain size.
+
+	It will first try to expand on both the lower and upper bound, and if not possible, it will expand the lower bound, and if that is not possible, it will expand the upper bound. If the resulting range is larger than the given `wholeRange`, it will be clamped to `wholeRange`.
+
+	- Precondition: The range and the given range must be positive.
+	- Precondition: The range must be a subset of the given range.
+
+	```
+	(1...1.2).minimumRangeDiff(of: 1, in: 0...4)
+	//=> 0.5...1.7
+
+	(0...0.5).minimumRangeDiff(of: 1, in: 0...4)
+	//=> 0...1
+
+	(3.5...4).minimumRangeDiff(of: 1, in: 0...4)
+	//=> 3...4
+
+	(0...0.1).minimumRangeDiff(of: 1, in: 0...4)
+	//=> 0...1
+	```
+	*/
+	func minimumRangeDiff(of diff: Bound, in wholeRange: ClosedRange<Bound>) -> ClosedRange<Bound> {
```

Comment:
@sunshinejr I'm not happy with the naming `minimumRangeDiff` and `wholeRange`. PLeAsE HeLp. 🙏